### PR TITLE
Update tutorials to alpha.6

### PR DESCRIPTION
**Pull request by JoshOrndorff**
This WIP PR updates tutorials to alpha.6. It is incomplete because we got stuck while updating the private network tutorial and reported the issue in https://github.com/paritytech/substrate/issues/5654

### CODE REVIEW

File: tuts/pallet-in-own-crate/v2.0.0-alpha.6/index.md [hunk 1]
```diff
@@ -57,10 +50,28 @@ The beginning of the `Cargo.toml` now looks like:
...
+repository = 'https://github.com/substrate-developer-hub/substrate-pallet-template'
+version = '2.0.0-alpha.6'
+```
+
+### Compile the Template Pallet
+
+> If you have been following the steps of this tutorial and cloned the template pallet into the `pallets` directory of
+your node template, you will need to add the template pallet to the `workspace.members` array of the `Cargo.toml` file
+in the root of the node template directory; you will need to include the path and the name of the package, so add this
+element to the `members` array: `'pallets/test-pallet',`.
```
Comment (JoshOrndorff) [inline tuts/pallet-in-own-crate/v2.0.0-alpha.6/index.md:68]: Nice!

File: tuts/build-a-dapp/v2.0.0-alpha.6/pallet.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+---
+
+The Substrate runtime is composed of FRAME pallets. You can think of these pallets as individual
+pieces of logic that define what your blockchain can do! Substrate provides you with a number of
+pre-built pallets for use in FRAME-based runtimes.
+
+![Runtime Composition](../assets/runtime.png)
+
+For example, FRAME includes a
+[Balances](https://substrate.dev/rustdocs/v2.0.0-alpha.6/pallet_balances/index.html) pallet that controls
```
Comment (JoshOrndorff) [inline tuts/build-a-dapp/v2.0.0-alpha.6/pallet.md]: docs.rs

File: tuts/build-a-dapp/v2.0.0-alpha.6/pallet.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+the underlying currency of your blockchain by managing the _balance_ of all the accounts in your
+system.
+
+If you want to add smart contract functionality to your blockchain, you simply need to include the
+[Contracts](https://substrate.dev/rustdocs/v2.0.0-alpha.6/pallet_contracts/index.html) pallet.
+
+Even things like on-chain governance can be added to your blockchain by including pallets like
+[Democracy](https://substrate.dev/rustdocs/v2.0.0-alpha.6/pallet_democracy/index.html),
+[Elections](https://substrate.dev/rustdocs/v2.0.0-alpha.6/pallet_elections/index.html), and
+[Collective](https://substrate.dev/rustdocs/v2.0.0-alpha.6/pallet_collective/index.html).
```
Comment (JoshOrndorff) [inline tuts/build-a-dapp/v2.0.0-alpha.6/pallet.md]: docs.rs

File: tuts/build-a-dapp/v2.0.0-alpha.6/pallet.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+            // Emit an event that the claim was erased
+            Self::deposit_event(RawEvent::ClaimRevoked(sender, proof));
+        }
+    }
+}
+```
+
+> The functions you see here do not have return types explicitly stated. In reality they all return
+> [`DispatchResult`](https://substrate.dev/rustdocs/v2.0.0-alpha.6/frame_support/dispatch/type.DispatchResult.html)s.
```
Comment (JoshOrndorff) [inline tuts/build-a-dapp/v2.0.0-alpha.6/pallet.md]: docs.rs

File: tuts/create-your-first-substrate-chain/v2.0.0-alpha.6/background.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+* and more...
+
+These technologies span a huge breadth of computer science, and thus usually require expert teams to
+develop. As a result, most blockchain projects are _not_ developed from the ground up. Instead,
+these projects are forked from already existing blockchain repositories. For instance:
+
+* The Bitcoin repository was forked to create: Litecoin, ZCash, Namecoin, Bitcoin Cash, etc...
+* The Ethereum repository was forked to create: Quorem, POA Network, KodakCoin, Musicoin, etc...
+
+<!--TODO This image link is broken -->
```
Comment (JoshOrndorff) [inline tuts/create-your-first-substrate-chain/v2.0.0-alpha.6/background.md]: I believe this is fixed now so the TODO can be removed.

File: tuts/pallet-in-own-crate/v2.0.0-alpha.6/index.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+repository](https://help.github.com/en/articles/create-a-repo) and [push your
+pallet's code](https://help.github.com/en/articles/pushing-to-a-remote) to it.
+
+### Publishing on Crates.io
+Crates.io allows permissionless publishing. Learn the proceedure following their own guide about [publishing on crates.io](https://doc.rust-lang.org/cargo/reference/publishing.html)
+
+## Updating your Runtime's Dependecy
+With your pallet now published on GitHub, crates.io, or both,
+we can update your runtime to use the code that is published instead of a
+hard-coded file system path.
```
Comment (JoshOrndorff) [inline tuts/pallet-in-own-crate/v2.0.0-alpha.6/index.md:252]: I like this section. Given that we're ultimately updating the dependency to grab it from github or crates.io, I'm thinking it makes more sense to clone the pallet template somewhere other than inside the node template.

This doesn't have to be resolved in this PR. But if it isn't we should make a ticket for it.

File: tuts/start-a-private-network/v2.0.0-alpha.5/customchain.md [hunk 7]
```diff
@@ -1,99 +1,10 @@
 ---
 slug: customchain
 lang: en
-title: Creating Your Private Network
+title: Launch Your Private Network
 ---
-Now that each participant has their own keys generated, you're ready to start your own custom chain. In this section it is no longer required to use a single physical machine or a single binary.
```
Comment (JoshOrndorff) [inline tuts/start-a-private-network/v2.0.0-alpha.5/customchain.md:6]: Note to reviewers. This section is not removed entirely. It is split into its own file `customspec.md`.

**Review by JoshOrndorff — COMMENTED** [review]
This looks good to me with just a few comments to address. Would appreciate a review from another reviewer since I wrote half of this.

File: tuts/add-a-pallet/v2.0.0-alpha.6/index.md [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+            })? // <- Remove semi-colon
+            /*** Add This Block ***/
+            .with_rpc_extensions(|builder| -> Result<RpcExtension, _> {
+                use pallet_contracts_rpc::{Contracts, ContractsApi};
+                let mut io = jsonrpc_core::IoHandler::default();
+                io.extend_with(
+                ContractsApi::to_delegate(Contracts::new(builder.client().clone()))
+                );
+                Ok(io)
+            })?;
```
Comment (riusricardo) [inline tuts/add-a-pallet/v2.0.0-alpha.6/index.md]:
```suggestion
            .with_rpc_extensions(|builder| -> Result<IoHandler<sc_rpc::Metadata>, _> {
                let handler = pallet_contracts_rpc::Contracts::new(builder.client().clone());
                let delegate = pallet_contracts_rpc::ContractsApi::to_delegate(handler);

                let mut io = IoHandler::default();
                io.extend_with(delegate);
                Ok(io)
            })?;
```

File: tuts/add-a-pallet/v2.0.0-alpha.6/index.md [hunk 9]
```diff
@@ -0,0 +1,508 @@
...
+pallet-contracts-rpc = '0.8.0-alpha.6'
+sc-rpc = '2.0.0-alpha.6'
+```
+
+**`node/src/service.rs`**
+```rust
+macro_rules! new_full_start {
+	($config:expr) => {{
+        /*** Add This Line ***/
+        type RpcExtension = jsonrpc_core::IoHandler<sc_rpc::Metadata>;
```
Comment (riusricardo) [inline tuts/add-a-pallet/v2.0.0-alpha.6/index.md]:
```suggestion
        use jsonrpc_core::IoHandler;
```

File: tuts/add-a-pallet/v2.0.0-alpha.6/index.md (hunk 8 above)
Comment (riusricardo) [reply tuts/add-a-pallet/v2.0.0-alpha.6/index.md]: Small legibility improvement.

File: tuts/create-your-first-substrate-chain/v2.0.0-alpha.6/interact.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+If the number after `best:` is increasing, your blockchain is producing new blocks!
+
+## Start the Front End
+
+To interact with the local node we will use the Polkadot-js Apps user interface, often known as
+"Apps" for short. Despite the name, Apps will work with any Substrate-based blockchain including ours, not just Polkadot.
+
+In your web browser, navigate to [https://polkadot.js.org/apps](https://polkadot.js.org/apps/#/settings?rpc=ws://127.0.0.1:9944).
+
+On the `Settings` tab ensure that you are connected to a `Local Node` or `ws://127.0.0.1:9944`.
```
Comment (riusricardo) [inline tuts/create-your-first-substrate-chain/v2.0.0-alpha.6/interact.md:63]: How are we going to handle the change in the Weight type?
Transfers will not work on polkadot.js.org/apps unless we define `Weight as u32`

**Review by riusricardo — COMMENTED** [review]
LGMT. Only a tiny blocker in the first chain tutorial.

File: tuts/create-your-first-substrate-chain/v2.0.0-alpha.6/interact.md (hunk 10 above)
Comment (JoshOrndorff) [reply tuts/create-your-first-substrate-chain/v2.0.0-alpha.6/interact.md:63]: This is solved with a change in Apps https://github.com/polkadot-js/apps/pull/2577

In the future we will give the node template a proper spec version.

**Review by danforbes — APPROVED** [review]
Committed @riusricardo suggestions :heavy_check_mark:

**Pull request merged**